### PR TITLE
Update ldi dependency

### PR DESCRIPTION
**Pull request by sashadev-sky**
Fixes #1379 (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

[//]: # (To mark checkbox write 'x' within the square brackets)

* [ ] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [ ] code is in uniquely-named feature branch and has no merge conflicts 📁
* [ ] screenshots/GIFs are attached 📎 in case of UI updation
* [ ] ask `@publiclab/mapknitter-reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

============================
- Remove extra spaces from README.md
- Update deprecated syntax `assert_equal nil, connection` -> `assert_nil connection`
- Add `yarn config set ignore-engines true` to `.travis.yml`
- Add  "node": ">=10.0.0" to engines in package.json
- Bump ldi 0.11.6 -> 0.20.7 in package.json
- Update ldi related code (Map.js) for compatibility with new version (`L.RevertAction` -> `L.RestoreAction`)
- Remove extra spacing from (Map.js)
- Update schema.rb to include @icarito last migration
- Update schema.rb.example to reflect new schema.rb
============================

### CODE REVIEW

**Comment by sashadev-sky** [comment]
@jywarren fixed the tests but there is another issue trying to fix and not sure about.. on page refresh the toolbar UI breaks (toolbar location does not render exactly where image is) and there is an error in the console

**Comment by sashadev-sky** [comment]
@jywarren ok fixed! This is ready for review

File: app/assets/javascripts/mapknitter/Map.js
Comment (sagarpreet-chadha) [inline app/assets/javascripts/mapknitter/Map.js:389]:
```suggestion
    layers = layers.filter(image => !!image._url);
```

File: app/assets/javascripts/mapknitter/Map.js
Comment (sagarpreet-chadha) [inline app/assets/javascripts/mapknitter/Map.js:401]:
```suggestion
      var layer = layers.filter(l => l._url === warpable.srcmedium)[0];
```

File: app/assets/javascripts/mapknitter/Map.js
Comment (sagarpreet-chadha) [inline app/assets/javascripts/mapknitter/Map.js:403]:
```suggestion
      if (!layer) {
```

File: package.json
Comment (sagarpreet-chadha) [inline package.json:47]: Let's mention this in readme?

Comment (sashadev-sky) [reply package.json:47]: @sagarpreet-chadha this was actually a non essential change ie. can be removed and won't change anything. Serves more as a hint I think. What fixed the problem was telling yarn to ignore engines in `.travis.yml`

**Review by sagarpreet-chadha — APPROVED** [review]

**Comment by sashadev-sky** [comment]
@sagarpreet-chadha what should we do about all these queued builds? Can't even locate them in Travis. Maybe @gauravano or @jywarren can merge here for me?

**Comment by sashadev-sky** [comment]
also thank you for reviewing @sagarpreet-chadha @VladimirMikulic !

**Comment by cesswairimu** [comment]
Maybe we could try closing then reopening the PR

**Comment by cesswairimu** [comment]
Well, that didn't work :confused:

**Comment by sagarpreet-chadha** [comment]
@cesswairimu one Travis check passed successfully. We can force merge to unblock @sashadev-sky , however did we changed any config in Travis for mapknitter project? Can you check once? Thanks 😄

**Comment by cesswairimu** [comment]
@sagarpreet-chadha  we haven't changed the config file in a while...the dependabot PRs have been running okay

**Comment by sashadev-sky** [comment]
@cesswairimu @sagarpreet-chadha its just because i committed @sagarpreet-chadha suggestions, this is safe to force merge - i don't have the admin rights tho can one of you please do it for me :)

**Comment by cesswairimu** [comment]
@sashadev-sky I do not have the admin rights to force. I guess @jywarren might have them

**Comment by sashadev-sky** [comment]
@sagarpreet-chadha @cesswairimu super weird but the problem was because the suggestions I committed from @sagarpreet-chadha are orphaned I guess? See https://github.com/publiclab/mapknitter/commit/ee7bd009dc79bf4d8bb389a6f71a892eb5bf48cd "This commit does not belong to any branch on this repository."

I had to rebase over upstream main. No idea why and still don't get it after researching. Any clue? 

Anyway, queued builds are gone so merging

**Comment by sashadev-sky** [comment]
Im thinking maybe for some reason "Allow edits by maintainers" was unchecked

**Pull request merged**
